json/src/ser/value.rs

```diff
@@ -113,6 +121,12 @@ impl Serialize for AsNumbers<'_> {
                 for number in numbers {
                     if number.is_finite() {
                         ser.serialize_element(&number)?;
+                    } else if number.is_nan() {
+                        ser.serialize_element(&f64::NAN.to_string())?;
```

Comment:
We can skip these allocations by resolving the string inline. As long as we are careful about inserting correct string literals, it should be fine.
```suggestion
                        ser.serialize_element("NaN")?;
```

Reply:
Good callout - updated to use constants to avoid the to_string()